### PR TITLE
Add keystone authentication proposal

### PR DESCRIPTION
**Pull request by pytimer**
This proposal said that Harbor using the non-OIDC Keystone as identity provider. How to implement and using it.

This is my first send the proposal in harbor community, welcome to someone review it and point out if i missing something about it. Thanks.

### CODE REVIEW

**Comment by steven-zou** [comment]
@pytimer 

Thank you for being willing to involve in the Harbor community and submit this proposal to fill in the gap of doing authentication with the Keystone. The maintainers will review your proposal and give feedback. Let's work together with the community to make it happen.


@goharbor/all-maintainers 

Please review this proposal and give your feedback. Thanks!

**Comment by pytimer** [comment]
@steven-zou Thanks. I will wait the maintainers review completed.

File: proposals/Keystone_Authentication_Support.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+#### Administrator configure Keystone Endpoint and domain
+
+The administrator or Harbor should be able to configure the Keystone identity provider via UI portal or configuration API.
+
+The attributes of Keystone identity provider:
+
+* **keystone_endpoint**: The endpoint URL of the Keystone identity provider. It protocol only support http now.
+* **keystone_domain_name**: The domain of the Keystone identity provider. It is a required attribute.
+
+#### User login to UI protal with Keystone identity provider
```
Comment (reasonerjt) [inline proposals/Keystone_Authentication_Support.md]: Please elaborate how Harbor will interact with keystone to perform the authentication.

File: proposals/Keystone_Authentication_Support.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+Author: `pytimer`
+
+## Abstract
+
+It proposes an approach to enable Harbor to support non-OIDC Keystone authentication provider.
+
+## Background
+
+Harbor running in the Kubernetes cluster which using the Keystone as authentication, we hope the Kubernetes and Harbor using the same identity provider, so Harobr needs integration with keystone as identity provider.
```
Comment (reasonerjt) [inline proposals/Keystone_Authentication_Support.md]: Could you elaborate how k8s cluster integrate with keystone for authentication?  Is there any reference to help us understand?

File: proposals/Keystone_Authentication_Support.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+To support the [user cases](#User cases) in the below section, we add Keystone authentication in Harbor. Admin can configure an Keystone identity provider, after that, user can use this Keystone identity provider as authentication way to login the Harbor's UI Portal. Besides that, user can also use docker CLI.
+
+### User cases
+
+* User running Harbor in the Openstack.
+* User running Harbor in the Kubernetes which using the non-OIDC Keyston as authentication.
+
+### Process flow
+
```
Comment (reasonerjt) [inline proposals/Keystone_Authentication_Support.md]: Will this proposal cover how keystone user should interact with Harbor's API?

File: proposals/Keystone_Authentication_Support.md (hunk 2 above)
Comment (kofj) [reply proposals/Keystone_Authentication_Support.md]: Same question too.

Comment (pytimer) [reply proposals/Keystone_Authentication_Support.md]: I add k8s integration with keystone document links. I'm not sure if i should describe it detail in this proposal, if needs, i will add later.

File: proposals/Keystone_Authentication_Support.md (hunk 3 above)
Comment (pytimer) [reply proposals/Keystone_Authentication_Support.md]: Describe how user acessing the Harbor's API in `User onboarded via the Keystone authentication flow accessing the API` section.

File: proposals/Keystone_Authentication_Support.md (hunk 1 above)
Comment (pytimer) [reply proposals/Keystone_Authentication_Support.md]: @reasonerjt  I add the user login flow chart, is it ok or i should add other description?

**Comment by OrlinVasilev** [comment]
@pytimer @reasonerjt what is the status of this one? looks like obsolete

**Comment by OrlinVasilev** [comment]
[close https://github.com/goharbor/community/pull/96](https://trello.com/c/d5hqUguA/15-close-https-githubcom-goharbor-community-pull-96)

**Comment by OrlinVasilev** [comment]
proceeding with closing this as nobody took action on it for the last 2 years! If needed it can be reopened!

**Pull request closed without merging**